resources/js/components/ui/form/Checkbox.tsx

```diff
@@ -0,0 +1,58 @@
+import {Component, JSX, createEffect, splitProps} from "solid-js";
+import * as checkbox from "@zag-js/checkbox";
+import {normalizeProps, useMachine} from "@zag-js/solid";
+import {createMemo, createUniqueId} from "solid-js";
+import s from "./Checkbox.module.scss";
+import {ValidationMessages} from "./ValidationMessages";
+import {useFieldLabel} from "./FieldLabel";
+
+export interface CheckboxProps extends Omit<JSX.InputHTMLAttributes<HTMLInputElement>, "onChange"> {
+  name: string;
+  label?: string;
+  disabled?: boolean;
+  onChange?: checkbox.Context["onChange"];
+  indeterminate?: boolean;
```

Comment:
To bym wywalił na razie, bo tak samo - tym powinien zajmować się felte. Albo pozwala tristate checkbox albo nie, ale to jego rzecz zarządzać stanem, a tu najwyżej można jego stan tłumaczyć na stan checkboxa.